File: api-tugas/app/Models/Dosen.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Dosen extends Model
{
    protected $table = 'dosen_tabel';

    protected $primaryKey = 'NIP';

    public $incrementing = false;

    protected $keyType = 'string';
    
    protected $fillable = [
        'NIP',
        'nama',
        'fakultas',
        'prodi',
        'jabatan_akademik',
        'email',
        'no_telepon',
        'status_aktif'
    ];

    public function bimbingan()
    {
        return $this->hasMany(Bimbingan::class, 'nip_dosen', 'NIP');
    }

    public function capaianPrestasi()
    {
        return $this->hasMany(CapaianPrestasi::class, 'NIP', 'NIP');
    }
}

File: capaian_prestasi/lib/models/dosen.dart
class Dosen {
  final String nip;
  final String nama;
  final String? fakultas;
  final String? prodi;
  final String? jabatanAkademik;
  final String? email;
  final String? noTelepon;

  Dosen({
    required this.nip,
    required this.nama,
    this.fakultas,
    this.prodi,
    this.jabatanAkademik,
    this.email,
    this.noTelepon,
  });

  factory Dosen.fromJson(Map<String, dynamic> json) {
    return Dosen(
      nip: json['NIP'] ?? '',
      nama: json['nama'] ?? '',
      fakultas: json['fakultas'],
      prodi: json['prodi'],
      jabatanAkademik: json['jabatan_akademik'],
      email: json['email'],
      noTelepon: json['no_telepon'],
    );
  }
}
